pytorch_lightning/loggers/tensorboard.py

```diff
@@ -114,13 +119,18 @@ def log_dir(self) -> str:
         """
         # create a pseudo standard path ala test-tube
         version = self.version if isinstance(self.version, str) else f"version_{self.version}"
-        log_dir = os.path.join(self.root_dir, version)
+        sub_dir = self.sub_dir if isinstance(self.sub_dir, str) else ""
+        log_dir = os.path.join(self.root_dir, version, sub_dir)
```

Comment:
Should we check sub_dir doesn't contain `~` or `..`

Reply:
I think you can just use the expand options for os in every case here.

Reply:
@justusschock I am not exactly following. You mean adding the `os.path.expand` option to every logger `log_dir` ?

Reply:
```suggestion
        log_dir = os.path.join(self.root_dir, version, sub_dir)
        log_dir = os.path.expandvars(log_dir)
        log_dir = os.path.expanduser(log_dir)
```

I think this is what justus means, would also need tests for `~` and `$` in paths.